virtual-prestage-child.groovy

```diff
@@ -175,27 +180,28 @@ void updateDevices(targets, targetProperty = null) {
                     skip += notApplicable;
                 }
             }
-
-            // This code currently does nothing, since the function is only
-            // called for devices that are on.  If we ever want to uncomment
-            // this, we'll need to call it for devices with this capability as
-            // well.
-            //
-            // if( property == "level" ) {
-            //     def presettable = (targets - skip).findAll { it.hasCapability("LevelPreset") };
-            //     presettable*.presetLevel(primaryDevice.currentValue("level"));
-            //     skip += presettable;
-            // }
-
-            def value = primaryDevice.currentValue(property);
-            applicable -= skip;
-            if( value != null  && applicable ) {
-                debug("Setting ${applicable} ${property} to ${value}");
-                applicable.each{
-                    if( it.currentValue("switch") == "on" ) {
-                        it."${command}"(value);
-                        pauseExecution(meteringDelay ?: 0);
-                    }
+            applicable -= skip
+
+            def onDevices = applicable.findAll { it.currentValue("switch") == "on" }
+            def offDevices = applicable - onDevices
+
+            if( offDevices && property == "level" ) {
+                def presettable = offDevices.findAll { it.hasCapability("LevelPreset") };
+                debug("Prestaging ${presettable} ${property} to ${value}");
+                presettable*.presetLevel(value);
+            } else if( offDevices && property == "colorTemperature" ) {
+                // UGLY UGLY HACK: The Philips Hue Zigbee Driver by jonathanb will prestage CT when given a CT while off; but the Advanced
+                // Zigbee Bulb driver will turn the light on.  Clumsily identify the Hue driver by checking for a unique command it provides.
```

Comment:
Is there a way to know what driver is behind a given device?  Didn't find anything in the docs.

Reply:
Not that I've found, unfortunately. The "correct" way to do this is with a property equivalent to LevelPreset, but that would require driver action first.